### DISCUSSION TITLE
Why are all fields included in a SQL SELECT query?

### DISCUSSION THREAD

**Discussion by dpcamp** [prisma/prisma#11726] [Q&A]
I was troubleshooting an issue with a select query, and it just dawned on me - if my apollo/graphql query only requests a few fields, the MS SQL query from Prisma still is selecting all fields to be returned. 

Is there a reason for that and is there a way to change that behavior without writing a custom query?

**Comment by janpio** [comment]
Does your Prisma Client query also specify only the fields it needs?

**Reply by dpcamp** [reply]
That I'm not sure - how would I check for that? 

I'm just using findMany() in my code, then in the terminal I'm noticing the SELECT statement is querying for all fields in the table. 

There are cases where I would need all fields, but if a graphql query is only asking for 2-3 fields I wouldn't think Prisma would continue to query every field. But that's an assumptions on my part.

**Reply by janpio** [reply]
You need to tell `findMany` somehow which fields you require. It can not magically know what fields you know. Here is some documentation: https://www.prisma.io/docs/concepts/components/prisma-client/select-fields

**Reply by dpcamp** [reply]
Thanks that's helpful. 

So just for my understanding - for example, lets say if I have a table that has displayName, username, and firstName.  If I run a graphql query of 
`
query AllUserProfileUsernames {
  allUserProfiles {
    displayName
    username
  }
}
`
Prisma will query my DB with SELECT displayName, username, and firstName because it does not know what exactly the grapqhl query is, correct?

**Reply by janpio** [reply]
Yes. Prisma only knows what your code tells Prisma. If you ask the GraphQL query for only a specific field, but then your code calls `findMany` for all fields, then Prisma will happily get and return all fields. If you code takes the list of fields from the GraphQL query and also puts that into the Prisma Client query, it will only query these fields.

**Reply by MurzNN** [reply]
Keystone needs data not only for requested fields, but for all fields of each model, to pass them to access and hooks functions, so it requires them all in SQL query.

**Reply by MurzNN** [reply]
But will be good to have ability to exclude some fields (that don't needed for hooks and access checks) from SQL queries, or manually set list of fields that always loads additionally to requested.

**Reply by pantharshit00** [reply]
Hello, our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

If you will run `findMany` without any select fields, it will always select everything. You can use library to generate the select field from GraphQL `info` object: https://paljs.com/plugins/select

As per regards with the `exclude` capability, that will be addressed via client extensions. See https://github.com/prisma/prisma/issues/15074

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
